tests/test_semdedup.py

```diff
@@ -247,57 +276,420 @@ def get_reference_embeddings(
+            emb_by_clust_dir=os.path.join(
+                clustering_output_dir, "embs_by_nearest_center"
+            ),
+            id_column="id",
+            which_to_keep=which_to_keep,
+            sim_metric=sim_metric,
+            output_dir=semantic_extraction_output_dir,
+            embedding_column="embeddings",
+            batched_cosine_similarity=20,
+        )
+
+        # Call compute_semantic_match_dfs
+        semantic_cluster_level_dedup.compute_semantic_match_dfs()
+
+        output_samples_per_cluster = []
+        semdedup_pruning_tables_df = []
+        # Check content of semdedup_pruning_tables
+        for i in range(self.n_clusters):
+            cluster_i_path = os.path.join(
+                semantic_extraction_output_dir,
+                "semdedup_pruning_tables",
+                f"cluster_{i}.parquet",
+            )
+            assert os.path.exists(cluster_i_path)
+            df = pd.read_parquet(cluster_i_path)
+            output_samples_per_cluster.append(df.shape[0])
+            assert df.columns.tolist() == [
+                "id",
+                "max_id",
+                "cosine_sim_score",
+            ]
+            semdedup_pruning_tables_df.append(df)
+
+        semdedup_pruning_tables_df = pd.concat(semdedup_pruning_tables_df)
+        np.testing.assert_allclose(
+            sorted(output_samples_per_cluster),
+            self.n_samples_per_cluster,
+            rtol=0.02,  # be within 2%
+        )
+
+        # Call extract_dedup_data
+        semantic_cluster_level_dedup.extract_dedup_data(eps_to_extract=0.01)
+        # Check content of unique_ids
+        unique_ids_path = os.path.join(
+            semantic_extraction_output_dir, "unique_ids_0.01.parquet"
+        )
+        assert os.path.exists(unique_ids_path)
+        unique_ids_df = pd.read_parquet(unique_ids_path)
+        assert unique_ids_df.columns.tolist() == [
+            "id",
+            "cosine_dist_to_cent",
+            "cluster",
+        ]
+
+        # Check content of semdedup_pruning_table with the filter matches the unique_ids
+        semdedup_pruning_tables_df_filtered = semdedup_pruning_tables_df[
+            semdedup_pruning_tables_df["cosine_sim_score"] < 1 - 0.01
+        ]
+        assert len(semdedup_pruning_tables_df_filtered) == len(unique_ids_df)
+        assert set(semdedup_pruning_tables_df_filtered["id"].to_list()) == set(
+            unique_ids_df["id"].to_list()
+        )
+
+        # Check content of summary file
+        summary_path = os.path.join(
+            semantic_extraction_output_dir, "dedup_summary_0.01.csv"
+        )
+        assert os.path.exists(summary_path)
+        df = pd.read_csv(summary_path)
+        if which_to_keep == "hard":
+            _kept, _removed = 29, 1471
+        elif which_to_keep == "easy":
+            _kept, _removed = 5, 1495
+        else:
+            # random is not deterministic, so we skip this test
+            return
```

Comment:
There is no seed, etc. that we could add to have reproducible results?

Reply:
I was able to add it in https://github.com/NVIDIA/NeMo-Curator/pull/615

Hardcoded `random_state=42` however if we wish to it then we'll have to expose another random_state param